test_scripts/MobileProjection/Defects/3142_01_Absence_of_misplaced_OnVideoDataStreaming.lua

```diff
@@ -0,0 +1,84 @@
+---------------------------------------------------------------------------------------------------
+-- Issue: https://github.com/smartdevicelink/sdl_core/issues/3142
+--
+-- Steps:
+-- 1. Set StopStreamingTimeout = 3000 and VideoDataStoppedTimeout = 3000 in SDL .INI file
+-- 2. Start SDL, HMI, connect Mobile device
+-- 3. Register PROJECTION (App_1) and NAVIGATION (App_2) applications
+-- 4. Activate App_1 and start Video streaming
+-- 5. Deactivate App_1 (streaming still continue since app has STREAMABLE state)
+-- 6. Activate App_2 and start Video streaming
+-- 7. App_1 and App_2 continue streaming data withing 'StopStreamingTimeout' timeout
+-- SDL does:
+--   - switch streaming between apps and provides HMI with streaming data from App_2
+--   - not unregister App_1 since timeout is not yet expired
+-- 8. App_1 stops streaming within 'StopStreamingTimeout' timeout
+-- SDL does:
+--   - not send Navi.OnVideoDataStreaming(false) notification to HMI since App_2 still continue streaming
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("test_scripts/MobileProjection/Defects/common")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Apps Configuration ]]
+common.app.getParams(1).appHMIType = { "PROJECTION" }
+common.app.getParams(2).appHMIType = { "NAVIGATION" }
+
+--[[ Local Functions ]]
+local function stopStreaming(pAppId, pServiceId)
+  common.hmi.getConnection():ExpectRequest("Navigation.StopStream", { appID = common.app.getHMIId(pAppId)})
+  :Do(function(_, data) common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", { }) end)
+  common.mobile.getSession(pAppId):ExpectEndService(pServiceId)
+  :Do(function() common.mobile.getSession(pAppId):SendEndServiceAck(pServiceId) end)
+end
+
+local function startStreaming(pAppId, pServiceId)
+  common.mobile.getSession(pAppId):StartService(pServiceId)
```

Comment:
@GetmanetsIrina Since during switching of the streaming we're expecting two `OnVideoDataStreaming` notifications: 1st - true and 2nd - false and taking into account that ATF can't handle two expectations with the same name in two functions catching of both is implemented in `activateApp2` function